### DISCUSSION TITLE
Optimize power consumption for single image capture

### DISCUSSION THREAD

**Discussion by BojanSof** [HimaxWiseEyePlus/Seeed_Grove_Vision_AI_Module_V2#45] [Q&A] [answered]
I was playing around with the Groove Vision AI V2 board with OV5647 camera module using the code in this repo.
I am interested in simple application that can capture a single image on request.
The request for capturing an image should be received via some communication protocol, for example SPI or I2C.

I created sample code by adding `scenario_app` and scraping the models inference from the `tflm_fd_fm` sample, which:
1. Initializes the chip in same manner as the `tflm_fd_fm` sample
2. Keeps the chip in idle for some time
3. Powers the camera module, initializes the image subsystem
4. Capture single image
5. Power down the camera

I've observed current consumption of 22 mA in idle, while the camera was off (shut down using its power enable GPIO pin), which seems a bit high, probably due to chip default configuration (400 MHz clock, etc.).
The current draw was measured using USB current meter.

1. Is there a sample code which configures the chip for lower current consumption while idle, i.e. while waiting for SPI/I2C request for capturing image?
2. Is there any guide maybe to optimizing power consumption?

Thank you in advance.

**Comment by stevehuang82** [comment, marked as answer]
Seeed Grove Vision AI Module V2 is a development board that is not designed for low power consumption.
The OV5647 camera is not a low power module and even if you turn it off using its power enable GPIO pin, the power consumption is still high. Moreover, the power consumption of the CH343P (UART to USB) device is also very high.

You can use app_pmu_enter_sleep() to set Himax WE2 to enter power-down mode to save power while system idle.
When Himax WE2 enters power-down mode, you must trigger AON_GPIO0(PA0) (D0/A0 pin on the board) to wake up Himax WE2 and start capturing images.

If you want to measure the difference in power consumption of Himax WE2 in all-on mode and power-down mode,
You should input 3.3V directly from the 3V3/GND pin on the board and disconnect the USB cable from the PC (there is no 5V input), then measure the 3.3V power consumption of Himax WE2 and OV5647 to exclude the CH343P power consumption.

app_pmu_enter_sleep() function code please refer to
https://github.com/HimaxWiseEyePlus/Seeed_Grove_Vision_AI_Module_V2/blob/main/EPII_CM55M_APP_S/app/scenario_app/allon_sensor_tflm_freertos/sleep_mode.c#L123

```
/**

* This function sets TIMER ID 2 or AON_GPIO0(PA0)/AON_GPIO1(PA1) as Power Down mode wake up sources.

* This function supports memory retention or memory no retention.

*

* @param timer_ms The wake up timer(millisecond).

* @param aon_gpio The index of the AON_GPIO pin to set as wake up source.

*                            0: AON_GPIO0(PA0), 1: AON_GPIO1(PA1), 0xFF: disable aon_gpio wake up.

* @param retention Memory retention or not.

*

*/

void app_pmu_enter_sleep(uint32_t timer_ms, uint32_t aon_gpio, uint32_t retention);

 

app_pmu_enter_sleep(0, 0, 0);         // AON_GPIO0(PA0) wake up, memory no retention
```

**Reply by BojanSof** [reply]
Thank you very much for your response. This is very helpful for my current benchmark.
